http-client/src/main/java/io/airlift/http/client/StreamingResponse.java

```diff
@@ -0,0 +1,6 @@
+package io.airlift.http.client;
+
+public interface StreamingResponse
+        extends Response, AutoCloseable
```

Comment:
Let's make this `extends Closeable` and override `close()` here to not throw any checked exception. The production implementation doesn't throw checked exceptions, and it doesn't seem useful since the caller can't do anything with the exception (it means some cleanup failed after the caller is done with the response).

Thinking about this more: under what cases would close throw?

Reply:
It's OK. For example, in the test the Guava `Closer` can throw. But we can wrap that in `UncheckedIOException`